cstar/roms/runtime_settings.py

```
@@ -0,0 +1,806 @@
+                case float():
+                    section_values_as_list_of_str.append(_format_float(value))
+                case Path():
+                    section_values_as_list_of_str.append(_format_path(value))
+                case None:
+                    continue
+                case _:
+                    section_values_as_list_of_str.append(_format_other(value))
+        section_values_as_single_str = self.value_joiner.join(
+            section_values_as_list_of_str
+        )
+
+        section_header = f"{self.section_name}: {' '.join(data.keys())}\n"
+        # Build the serialized string
+        serialized = ""
+        serialized += section_header
+        serialized += f"    {section_values_as_single_str}\n"
+        serialized += "\n"
+        return serialized
+
+    @classmethod
+    def from_lines(
+        cls, lines: Optional[list[str]]
+    ) -> Optional["ROMSRuntimeSettingsSection"]:
+        """This takes a list of lines as would be found under the section header of a
+        roms.in file and returns a ROMSRuntimeSettingsSection instance.
+
+        It uses the typehints of the ROMSRuntimeSettingsSection subclass under consideration to map
+        values from entries in the lines to correctly typed values.
+
+        If the type of an entry is "list", this is either the sole or final entry,
+        so we add all the remaining values to the list and break.
+
+        Parameters
+        ----------
+        lines: list[str]
+           Raw lines taken from a UCLA-ROMS runtime settings (`.in`) file.
+
+        Examples
+        --------
+        >>> LinRhoEos.from_lines(["0.2 1.0 0.822 1.0"])
+            LinRhoEos(Tcoef=0.2, T0=1.0, Scoef=0.822, S0=1.0)
+
+        >>> InitialConditions.from_lines(["1", "input_datasets/roms_ini.nc"])
+            InitialConditions(nrrec=1, ininame=Path("input_datasets/roms_ini.nc"))
+        """
+        if not lines:
```

this should raise an Exception. a file that should have content does not and the simulation should not be expected to behave correctly without the data.